nixos/modules/virtualisation/containers.nix

```diff
@@ -318,6 +319,16 @@ let
       '';
     };
 
+    hostPort = mkOption {
+      type = types.nullOr types.string;
```

Comment:
I'd make it a submodule, like this (with missing `example`/`description` attrs):
```nix
{
  type = types.listOf (types.submodule {
    options = {
      protocol = mkOption {
        type = types.str;
        default = "tcp";
      };
      hostPort = mkOption {
        type = types.int;
      };
      containerPort = mkOption {
        type = types.nullOr types.int;
        default = null;
      };
    };
  });
}
```